core/src/main/java/io/snappydata/impl/SnappyHiveCatalog.java

```diff
@@ -180,18 +201,19 @@ public Table getTable(String schema, String tableName, boolean skipLocks) {
 
   @Override
   public boolean isColumnTable(String schema, String tableName, boolean skipLocks) {
-    HMSQuery q = getHMSQuery();
-    q.resetValues(HMSQuery.ISCOLUMNTABLE_QUERY, tableName, schema, skipLocks);
-    Future<Object> f = this.hmsQueriesExecutorService.submit(q);
-    return (Boolean)handleFutureResult(f);
+    return isColumnTable(getTable(schema, tableName, skipLocks));
   }
 
   @Override
-  public boolean isRowTable(String schema, String tableName, boolean skipLocks) {
-    HMSQuery q = getHMSQuery();
-    q.resetValues(HMSQuery.ISROWTABLE_QUERY, tableName, schema, skipLocks);
-    Future<Object> f = this.hmsQueriesExecutorService.submit(q);
-    return (Boolean)handleFutureResult(f);
+  public boolean isColumnTable(Object hiveTable) {
+    return !ExternalTableType.getTableType((Table)hiveTable).equalsIgnoreCase(
```

Comment:
What happens if isColumnTable is called for an external table. !Row will make it column?

Reply:
Yeah, not sure why this was done in previous code. Tracking it down, this should be true for column and sample, but since earlier we had only row, column and sample so perhaps this made sense. Will change to be column+sample.

Reply:
Column+sample+index